// @canvasjs/angular-charts/esm2015/lib/canvasjs-angular-charts.module.js
import { NgModule } from '@angular/core';
import { CommonModule } from '@angular/common';
import { CanvasJSChart } from './canvasjs-angular-charts.component';
import * as i0 from "@angular/core";
export class CanvasJSAngularChartsModule {
}
CanvasJSAngularChartsModule.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.17", ngImport: i0, type: CanvasJSAngularChartsModule, deps: [], target: i0.ɵɵFactoryTarget.NgModule });
CanvasJSAngularChartsModule.ɵmod = i0.ɵɵngDeclareNgModule({ minVersion: "12.0.0", version: "12.2.17", ngImport: i0, type: CanvasJSAngularChartsModule, declarations: [CanvasJSChart], imports: [CommonModule], exports: [CanvasJSChart] });
CanvasJSAngularChartsModule.ɵinj = i0.ɵɵngDeclareInjector({ minVersion: "12.0.0", version: "12.2.17", ngImport: i0, type: CanvasJSAngularChartsModule, imports: [[
            CommonModule
        ]] });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.17", ngImport: i0, type: CanvasJSAngularChartsModule, decorators: [{
            type: NgModule,
            args: [{
                    declarations: [CanvasJSChart],
                    imports: [
                        CommonModule
                    ],
                    exports: [CanvasJSChart]
                }]
        }] });
